### PR TITLE
226 column control style for sections

### LINKED ISSUE #226: L1 Column-Control style for Sections
Figma: https://www.figma.com/file/oSetT4LbatRmXlcB2A7V8Z/Content-Hubs-2024?type=design&node-id=3797-102080&mode=design&t=YJDSOoj8I5PyddrZ-4

- (2) is a separate hero,
- the rest is a section, specially styled
- (22A) are content cards #146
- (25B) is a styled content list #150

- [ ] create section style, e.g. 2-1-column
- [ ] if required, handle block styling for contained blocks

Image

### PR DESCRIPTION
**Pull request by rootpea**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #226

Test URLs:
- Before: https://main--hlx-test--urfuwo.hlx.live/draft/rootpea/226-article-draft-rootpea
- After: https://226-column-control-style-for-sections--hlx-test--urfuwo.hlx.page/draft/rootpea/226-article-draft-rootpea

- [x] New Blocks introduced in this PR
      If yes, please provide details below

Block Name(Section Style Name)    | Documentation | Library Link
------------- | -------------|----------------
column-section | [doc-link](https://sap.sharepoint.com/:w:/r/sites/207899/Shared%20Documents/website/tools/sidekick/blocks/column-section.docx?d=wa06842ecba2a4741bf3b2012739ed336&csf=1&web=1&e=Gh4aaZ) | [Library Link](https://226-column-control-style-for-sections--hlx-test--urfuwo.hlx.page/tools/sidekick/blocks/column-section)

- [ ] New variations introduced in this PR

### CODE REVIEW

**Comment by rootpea** [comment]
Block Name(Section Style Name)    | Documentation | Library Link
------------- | -------------|----------------
column-section | [doc-link](https://sap.sharepoint.com/:w:/r/sites/207899/Shared%20Documents/website/tools/sidekick/blocks/column-section.docx?d=wa06842ecba2a4741bf3b2012739ed336&csf=1&web=1&e=Gh4aaZ) | [Library Link](https://226-column-control-style-for-sections--hlx-test--urfuwo.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/tools/sidekick/blocks/column-section&index=1)

File: scripts/scripts.js [hunk 1]
```diff
@@ -144,6 +144,33 @@ async function decorateVideoLinks(main) {
...
+        left.append(e.cloneNode(true));
+      } else {
+        right.append(e.cloneNode(true));
+      }
+    });
+    section.append(left);
+    section.append(right);
+    section.classList.add('column-section');
+  });
+}
```
Comment (benpeter) [inline scripts/scripts.js:172]: this is a little more concise:
```suggestion
function decorateMultiColumnSections(main) {
  main.querySelectorAll(':scope > div[class^="column-section-"]').forEach((section) => {
    const left = document.createElement('div'), right = document.createElement('div');
    left.className = 'column-section-left-block column-section-block';
    right.className = 'column-section-right-block column-section-block';

    Array.from(section.children).forEach((e) => {
      (e.classList.contains('right-style-wrapper') ? right : left).append(e.cloneNode(true));
    });

    section.append(left, right);
    section.classList.add('column-section');
  });
}
```

File: scripts/aem.js [hunk 2]
```diff
@@ -621,6 +621,12 @@ function decorateBlock(block) {
     block.dataset.blockStatus = 'initialized';
     const blockWrapper = block.parentElement;
     blockWrapper.classList.add(`${shortBlockName}-wrapper`);
+    for (let i = 1; i < block.classList.length; i += 1) {
```
Comment (benpeter) [inline scripts/aem.js:624]: do we ever change `aem.js`? I always thought this is off-limits :)

**Review by benpeter — COMMENTED** [review]
LGTM just unsure about changing aem.js and one suggestion to make the code more concise.

File: scripts/aem.js (hunk 2 above)
Comment (rootpea) [reply scripts/aem.js:624]: We do change it, there should be no issue adding functionality here. 👍

File: scripts/scripts.js [hunk 3]
```diff
@@ -144,6 +144,26 @@ async function decorateVideoLinks(main) {
   });
 }
 
+/**
+ * Decorates all multi-column sections in a container element.
+ * @param {Element} main The container element
+ */
+function decorateMultiColumnSections(main) {
+  main.querySelectorAll(':scope > div.column-section-1-1, :scope > div.column-section-3-2, :scope > div.column-section-2-3, :scope > div.column-section-2-1, :scope > div.column-section-1-2, :scope > div.column-section-3-1, :scope > div.column-section-1-3').forEach((section) => {
+    const left = document.createElement('div');
```
Comment (pahupe) [inline scripts/scripts.js:153]: Just a suggestion: we have been using the dom-builder.js to create DOM nodes and add CSS classes to them, as in: _div({ class: 'search-e' })_

**Review by pahupe — APPROVED** [review]
LGTM

**Pull request merged**